### PR TITLE
Profile model

### PR DESCRIPTION
**Pull request by Gamey001**
### What this PR does (required):
- stores users profiles

### Screenshots / Videos (required):
- 
profModel1

profModel2

### Any information needed to test this feature (required):
- To you test this  feature, you need to use the create-profile controller or try to create a record in the profiles collection on mongodb Atlas
### Any issues with the current functionality (optional):
- none

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -25,3 +25,4 @@ __snapshots__
 
 # debug & other vs code settings
 .vscode
+*.env
```
Comment (moffatethan) [inline .gitignore:28]: We don't want to ignore the sample env, I would revert this change please.

File: server/db.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 const mongoose = require("mongoose");
+require('dotenv').config({ path: "./sample.env" });
```
Comment (moffatethan) [inline server/db.js:2]: I believe this is already included in another file and wouldn't be required here.

Comment (moffatethan) [reply server/db.js:2]: Also, the `sample.env` is an example env so for future collaborators in a project can know what variables are needed in their environment to spin up the application.

The idea is for you to copy the `sample.env` and make a file called `.env`

File: server/models/Profile.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    type: String,
+    required: true,
+  },
+  gender: {
+    type: String,
+    enum: ['MALE', 'FEMALE', 'OTHER'],
+    required: true,
+    default: 'MALE'
+  },
+  birthDate: {
```
Comment (moffatethan) [inline server/models/Profile.js:18]:
```suggestion
  birthday: {
```

File: server/models/Profile.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    required: true
+  },
+  description: {
+    type: String,
+    required: true
+  },
+  availability: [{
+    type: Date, 
+    required: true
+  }],
```
Comment (moffatethan) [inline server/models/Profile.js:42]: Could you elaborate on the decision here a bit more? An array of dates, thinking of the front-end here. Would this mean the user needs to input every day they are available. 

Here are my thoughts. What if we make a general availability `Mon, Tues, Wed, Thurs, Fri, etc` and allow the user to select which days they are available as well as a time range of availability. UI/UX wise this is a lot more efficient. Thoughts?

**Review by moffatethan — CHANGES_REQUESTED** [review]
Overall good, nothing too concerning. Just revert that sample.env ignore and **make sure to clear your sample env from the tokens so we aren't exposing anything.**

Create a `.env` on your branch and everything should work just fine.

File: .gitignore (hunk 1 above)
Comment (Gamey001) [reply .gitignore:28]: noted with thanks

File: server/db.js (hunk 2 above)
Comment (Gamey001) [reply server/db.js:2]: noted with thanks

File: server/models/Profile.js (hunk 3 above)
Comment (Gamey001) [reply server/models/Profile.js:18]: noted with thanks

File: server/db.js (hunk 2 above)
Comment (Gamey001) [reply server/db.js:2]: Although I had an issue with the server refusing to read the MONGO_URI env variable. The server could not read the variable until I specified the path as you saw. Notwithstanding, I will located the file where the `dotenv` package was required and see if I can fix the issue and discard my line. Thanks so much.

File: server/models/Profile.js (hunk 4 above)
Comment (Gamey001) [reply server/models/Profile.js:42]: Sounds good Sir. I was actually thinking of making it in such a way that we can receive the availability through the HTML Datetime input and store in the availability field and extracting the date and time, but I get your point (we should be working with' week days' and not 'dates'. So I will the change the field to an array of objects, with `day` and `time range` properties?

**Review by moffatethan — APPROVED** [review]
LGTM!
